openadapt/privacy/providers/private_ai.py

```diff
@@ -0,0 +1,191 @@
+            "X-API-KEY": config.PRIVATE_AI_API_KEY,
+        }
+
+        response = requests.post(BASE64_URL, json=payload, headers=headers)
+        response = response.json()
+        logger.debug(f"{response=}")
+
+        # According to the API documentation,
+        # the response is a list of dicts when there is no error/issue in the request
+        # else it is a dict with a key "detail" containing the error message
+
+        if type(response) is dict and "detail" in response:
+            raise ValueError(response.get("detail"))
+
+        redact_file_path = os.path.join(FILES_DIR, f"redacted-{TEMP_IMAGEFILE_NAME}")
+
+        # Write to file
+        with open(redact_file_path, "wb") as redacted_file:
+            processed_file = response.get("processed_file").encode("ascii")
+            processed_file = base64.b64decode(processed_file, validate=True)
+            redacted_file.write(processed_file)
+
+        # return PIL.IMAGE type of redacted image
+        with open(redact_file_path, "rb") as file:
+            redacted_file_data = file.read()
+        os.remove(redact_file_path)  # remove redacted image after reading data
+
+        redact_pil_image_data = Image.open(BytesIO(redacted_file_data))
+
+        return redact_pil_image_data
+
+    def scrub_pdf(self, path_to_pdf: str) -> str:
+        """Scrub the PDF of all PII/PHI.
+
+        Args:
+            path_to_pdf (str): Path to the PDF to be redacted
+
+        Returns:
+            str: Path to the redacted PDF
+        """
+        # Read from file
+        with open(path_to_pdf, "rb") as b64_file:
+            file_data = base64.b64encode(b64_file.read())
+            file_data = file_data.decode("ascii")
+
+        payload = {
+            "file": {"data": file_data, "content_type": PDF_CONTENT_TYPE},
+            "entity_detection": {"accuracy": "high", "return_entity": True},
+            "pdf_options": {"density": 150, "max_resolution": 2000},
+            "audio_options": {"bleep_start_padding": 0, "bleep_end_padding": 0},
+        }
+
+        headers = {
+            "Content-Type": HEADER_CONTENT_TYPE,
+            "X-API-KEY": config.PRIVATE_AI_API_KEY,
+        }
+
+        response = requests.post(BASE64_URL, json=payload, headers=headers)
+        response_data = response.json()
+
+        # According to the API documentation,
+        # the response is a list of dicts when there is no error/issue in the request
+        # else it is a dict with a key "detail" containing the error message
+
+        if isinstance(response_data, dict) and "details" in response_data:
+            raise ValueError(response_data.get("detail"))
+
+        logger.debug(f"{response_data.get('entities')=}")
+        logger.debug(f"{len(response_data.get('entities'))=}")
+
+        redacted_file_path = path_to_pdf.split(".")[0] + "_redacted.pdf"
+
+        # Write to file
+        with open(redacted_file_path, "wb") as redacted_file:
```

Comment:
What do you think about using `BytesIO` here?

Reply:
Addressed in :[2f92260](https://github.com/OpenAdaptAI/OpenAdapt/pull/486/commits/2f922602e139e32bead068caf961238b23fb3b6a)